### DISCUSSION TITLE
Is there a reason for not allowing access to directories above the Acornfile directory?

### DISCUSSION THREAD

**Discussion by sbmw** [acorn-io/runtime#1952] [General]
Take the following local directory structure.  Where Deployment and Service1 are git projects.

├── Repos
│   ├── ProjectA
│   │   ├── Deployment
│   │   │   ├── Acornfile.local
│   │   ├── Services
│   │   │   ├── Service1
│   │   │   │   ├── src

During local development, we would like to be able to do things like reference projects within the group.
```
dirs: {
    "/app": "../Services/Service1"
}
```
or projects elsewhere in the /repo tree
```
dirs: {
    "/foo": "../../CommonLibProject"
}
```
or even something like ~/.npmrc
```
dirs: {
    "/app/someconfigfile.json": "/home/someconfigfile.json"
}
```
However, each of the above would be disallowed by (I think!) https://github.com/acorn-io/aml/blob/7aded86d1c3ec20d6bfe169880a44a5fc3a049da/schema/v1/app.cue#L168

So, my question: is there a reason for disallowing references outside of the project route or can the regex that validates directory paths be made less strict?

**Comment by ibuildthecloud** [comment]
This is just a technical issue.  The use case make sense. It has to do with the fact that the path of the files reference is technically relative to the `build.context` field.  I'm not sure, but guess this might work if you do `build: context: "../.."`.  That will then make the files and the dockerfile build relative to the parent directory.  I'm not 100% sure that works, but the technical theory lines up :)

So basically what you would do
```cue
containers: foo: {
  build: {
     dockerfile: "Dockerfile"
     context: "../.."
  }
  files: "/foo": "CommonLibProject"
}
```
Then in your `Dockerfile` all `COPY` statements must be relative to `../..`.  If this solution works for you, we can fix any bugs you might hit (again in theory should work, but I think we have weird pathing issues atm).

**Comment by sbmw** [comment]
Hi, thanks very much for getting back to me.  For us at least, setting eg `build: context: "../.."` would be fine, but unfortunately it doesn't seem to work in my tests.  I can close this discussion and raise an issue with a simple reproduction repo if that's the best way forward?

**Comment by sbmw** [comment]
I think I was wrong on this.  Allowing directory sync from any location might be nice, but perhaps isn't the acorn way.

The "access projects elsewhere" example
```
dirs: {
    "/foo": "../../CommonLibProject"
}
```
Can instead be handled by nesting that project's own Acornfile:
```
acorns: {
    "common-service-acorn": {
        build: {
            context: "../../CommonProject"
            acornfile: "../../CommonProject/Acornfile"
        }
        publishMode: "defined"
        profiles: "dev"
    }
}
```
And, "quick access to user config files during dev"
```
dirs: {
    "/app/someconfigfile.json": "/home/someconfigfile.json"
}
```
Should probably be handled with secrets
```
files: {
    "/home/.npmrc": "secret://npm-config/file"
}
```
With the first example of being able to have the Acornfile somewhere other than the top-level.  That's not so straightforward, but we can easily workwround by symlinking the Acornfile to the top-level with a project init script.   

So I'm going to close this as I have all the answers I need.  Thank you for pointing me in the right direction.
